### PR TITLE
Add allow-list config option to securitychecker_enlightn

### PR DESCRIPTION
**Pull request by peterjaap**
| Q             | A
| ------------- | ---
| Branch        | master for features and deprecations
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Documented?   | yes
| Fixed tickets | https://github.com/phpro/grumphp/discussions/1037

Add `allow-list` config option to be able to ignore given vulns, see https://github.com/enlightn/security-checker#allow-vulnerabilities

### CODE REVIEW

**Comment by peterjaap** [comment]
Needs some work, I'll get back to this.

**Comment by peterjaap** [comment]
@veewee done! :partying_face:

File: src/Task/SecurityCheckerEnlightn.php [hunk 1]
```diff
@@ -45,6 +47,9 @@ public function run(ContextInterface $context): TaskResultInterface
         $arguments = $this->processBuilder->createArgumentsForCommand('security-checker');
         $arguments->add('security:check');
         $arguments->addOptionalArgument('%s', $config['lockfile']);
+        foreach ($config['allow-list'] as $cve) {
```
Comment (veewee) [inline src/Task/SecurityCheckerEnlightn.php:50]: Wouldn't it be better to 
```php
->addArgumentArray('--allow-list=%s', $config['allow-list'])
```
or is there a specific reason you want to add an optional argument?

File: test/Unit/Task/SecurityCheckerEnlightnTest.php [hunk 2]
```diff
@@ -27,6 +27,7 @@ public function provideConfigurableOptions(): iterable
             [
                 'lockfile' => './composer.lock',
                 'run_always' => false,
+                'allow-list' => [],
             ]
```
Comment (veewee) [inline test/Unit/Task/SecurityCheckerEnlightnTest.php:31]: Can you also add test-cases covering this new argument?

**Review by veewee — CHANGES_REQUESTED** [review]
Thanks for the PR!

It seems like it is not completely ready yet. Can you take a look at the remarks?

File: src/Task/SecurityCheckerEnlightn.php (hunk 1 above)
Comment (peterjaap) [reply src/Task/SecurityCheckerEnlightn.php:50]: I've never heard of `addArgumentArray`, I can't find documentation on `addArgumentArray` anywhere?

File: test/Unit/Task/SecurityCheckerEnlightnTest.php (hunk 2 above)
Comment (peterjaap) [reply test/Unit/Task/SecurityCheckerEnlightnTest.php:31]: Like which cases?

**Comment by veewee** [comment]
@peterjaap It's a nice feature... Care to finish up this PR? :)

**Comment by peterjaap** [comment]
@veewee sure, but I've asked you some questions in the comments

**Comment by veewee** [comment]
> @veewee sure, but I've asked you some questions in the comments

@peterjaap 
Is it possible you made those comments as part of a code review but did not submit those comments? I am not seeing any.

**Comment by peterjaap** [comment]
@veewee woops, you're right :see_no_evil: Almost a year ago, lol.

File: src/Task/SecurityCheckerEnlightn.php (hunk 1 above)
Comment (veewee) [reply src/Task/SecurityCheckerEnlightn.php:50]: In your current code, you are iterating over the allow-list configuration just to add it as an argument;
Inside the `$arguments` (of type `\GrumPHP\Collection\ProcessArgumentsCollection`), there is a shortcut method `addArgumentArray()` that does this for you.
This is being used all over the repository to make the tasks easier to read.

You could use it like this:

```php
->addArgumentArray('--allow-list=%s', $config['allow-list'])
```

File: test/Unit/Task/SecurityCheckerEnlightnTest.php (hunk 2 above)
Comment (veewee) [reply test/Unit/Task/SecurityCheckerEnlightnTest.php:31]: The test cases should cover all options : you currently changed the default test-case, which has an empty allow list.
On top of that, a new case must be added that tests that if you pass a non-empty array to the `allow-list`, that the resulting CLI command contains one or multiple `--allow-list` flags

**Comment by veewee** [comment]
We notices this PR became inactive. If you wish to continue, feel free to pick it up again.
If not, we will be closing this PR soon. Thanks for your understanding.

**Comment by veewee** [comment]
Duplicate of newer PR #1161
